tests/test_core.py

```diff
@@ -4631,6 +4631,42 @@ def test_dylink_weak(self):
     side = test_file('core', 'test_dylink_weak_side.c')
     self.dylink_testf(main, side, force_c=True, need_reverse=False)
 
+  @node_pthreads
+  @needs_dylink
+  def test_dylink_tls(self):
+    # We currently can't export TLS symbols from module since we don't have
+    # and ABI for signaling which exports are TLS and which are regular
+    # data exports.
+
+    # TODO(sbc): Add tests that depend on importing/exported TLS symbols
+    # once we figure out how to do that.
+    create_file('main.c', r'''
+      #include <stdio.h>
+
+      _Thread_local int foo = 10;
```

Comment:
Is C++ `thread_local` implemented any differently from C `_Thread_local`? If so, do we have tests for that anywhere in emscripten?

Reply:
I assume its the same.      C11's `threads.h` header provides this macro:

`#define thread_local _Thread_local  (since C11)`

Which I assume is the to be compatible with the C++ keyword.  I can switch that if you find it more readable?